### ISSUE TITLE
Type annotation with try-imports does not work.

### ISSUE THREAD

**Issue by NCBM** [microsoft/pyright#4126]
**Describe the bug**
I have a piece of code below with try-import structure. It runs as expect,
but errors under pylance with `Illegal type annotation`.

**To Reproduce**
1. Make try-imports with types and type aliases;
2. Write imported class (which are all types and type aliases) in annotation.

**Expected behavior**
Pylance doesn't report `Illegal type annotation`.

**Screenshots or Code**
```python
import tkinter as tk

# attempt to use PIL
try:
    from PIL.Image import Image
    from PIL.ImageTk import PhotoImage
    HAS_PIL = True
except ImportError:
    Image = tk.Image
    PhotoImage = tk.PhotoImage
    # from tkinter import Image, PhotoImage  # this make the code below works.
    HAS_PIL = False


class Widget(...):
    ...

    @property
    def image(self) -> PhotoImage:  # type: ignore
        """The image of the widget."""
        return self.base["image"]

    @image.setter
    def image(self, img: Union[str, Image, PhotoImage]):  # the line that reports error
        _img = (
            PhotoImage(file=img)
                if isinstance(img, str) else
            PhotoImage(img)  # type: ignore
                if isinstance(img, Image) and HAS_PIL else
            img
        )
        self._image_slot = _img
        self.base["image"] = _img
```

**VS Code extension or command-line**
Running pylance `v2022.10.10` as VS Code extension.

**Additional context**
None.

**Comment by erictraut** [comment]
A type checker cannot determine statically whether the `try` or `except` will be executed at runtime. If you want to use the "try import" pattern, you can use `TYPE_CHECKING` to disambiguate. You'll need to decide whether you want the type checker to use the `PIL.Image` or `tk.Image` for its type analysis. Here's what it would look like if you want it to use the former:

```python
if TYPE_CHECKING:
    from PIL.Image import Image
    from PIL.ImageTk import PhotoImage
else:
    # attempt to use PIL
    try:
        ...
```
